src/msg/producer/writer/shard_writer.go

```diff
@@ -160,6 +160,12 @@ func newReplicatedShardWriter(
 
 func (w *replicatedShardWriter) Write(rm *producer.RefCountedMessage) {
 	w.RLock()
+	if len(w.messageWriters) == 0 {
+		w.RUnlock()
+		w.logger.Error("no message writers available for shard", zap.Int("shard", int(rm.Shard())))
```

Comment:
Nit: why not avoid type cast here by using `zap.Uint32`?

Reply:
https://github.com/m3db/m3/pull/3558/commits/808c4b708620664e7a8bc5a9d753aa1df7864b9a#diff-9a07267e28d677da0a5b4e38b8fe6509ae008c793bb58a92c4d2cc24ccda51c0R166